content/en/blog/_posts/2025-07-28-kubernetes-v1.34-sneak-peek.md

```diff
@@ -0,0 +1,97 @@
+This effort is complemented by a parallel enhancement, [KEP-647](https://kep.k8s.io/647), which brings the same tracing capabilities to the Kubernetes API server. Together, these enhancements provide a more unified, end-to-end view of events, simplifying the process of pinpointing latency and errors from the control plane down to the node. These features have matured through the official Kubernetes release process. [KEP-2831](https://kep.k8s.io/2831) was introduced as an alpha feature in v1.25, while [KEP-647](https://kep.k8s.io/647) debuted as alpha in v1.22. Both enhancements were promoted to beta together in the v1.27 release. Looking forward, Kubelet Tracing ([KEP-2831](https://kep.k8s.io/2831)) is now targeting graduation to a stable in the upcoming v1.34 release.
+
+### PreferSameZone and PreferSameNode Traffic Distribution for Services
+
+The `spec.trafficDistribution` field within a Kubernetes Service allows users to express preferences for how traffic should be routed to Service endpoints.  
+[KEP-3015](https://kep.k8s.io/3015) deprecates `PreferClose` and introduces two additional values: `PreferSameZone` and `PreferSameNode`. `PreferSameZone` is equivalent to the current `PreferClose`. `PreferSameNode` prioritizes sending traffic to endpoints on the same node as the client.  
+This feature was introduced in v1.33 behind the `PreferSameTrafficDistribution` feature gate. It is targeting graduation to beta in v1.34 with its feature gate enabled by default.
+
+### `kubectl` supports KYAML as an output format
+
+[KEP-5295](https://kep.k8s.io/5295) introduces KYAML, a safer and less ambiguous YAML subset and encoding, designed specifically for Kubernetes. KYAML addresses the shortcomings of both YAML and JSON. YAML's significant whitespace requires careful attention to indentation and nesting, while its optional string-quoting can lead to unexpected type coercion. Meanwhile, JSON lacks comment support and has strict requirements for trailing commas and quoted keys.  
+KYAML addresses the problems by:
+
+* Always double-quoting value strings   
+* Leaving keys unquoted unless they are potentially ambiguous  
+* Always using {} for structs and maps  
+* Always using [] for lists
+
+This might sound a lot like JSON, because it is! But unlike JSON, KYAML:
+
+* Supports comments  
+* Allows trailing commas in lists and maps for improved usability.  
+* Does not require quoted keys.
+
+In Kubernetes v1.34, KYAML will be introduced as a new output format for `kubectl` (as in `kubectl get -o kayml ...`). KYAML is and will remain a strict subset of YAML, ensuring that any compliant YAML parser can parse KYAML documents.
```

Comment:
```suggestion
In Kubernetes v1.34, you can use KYAML for manifests and in Helm charts and request KYAML output from`kubectl` (as in `kubectl get -o kayml ...`). KYAML is and will remain a strict subset of YAML, ensuring that any compliant YAML parser can parse KYAML documents.
```

Reply:
@lmktfy @thockin Please let me know if the above is the correct understanding. Also, given @thockin mentioned the delivery timeline, what information is appropriate to disclose here?

Reply:
Yes, KYAML is supposed to work with all yaml parsers (unless there is a bug in the encoder - which would be subsequently fixed)